src/indicators/drought/iri_rainfallforecast.py

```diff
@@ -43,44 +41,79 @@ def download_iri(iri_auth, config, chunk_size=128):
     # couldn't figure how to improve it
     logger.info("Downloading IRI NetCDF file. This might take some time")
     response = requests.get(config.IRI_URL, cookies=cookies, verify=False)
-    with open(IRI_filepath, "wb") as fd:
+    with open(output_path, "wb") as fd:
         for chunk in response.iter_content(chunk_size=chunk_size):
             fd.write(chunk)
 
-    # TODO: explore if can also open with rioxarray instead of xarray.
-    # Getting an error with current settings
+
+def clean_iri_data(
+    input_path: str,
+    output_path: str,
+    config: Config(),
+):
+    """
+    clean up some flaws of IRI's format and write the cleaned
+    data to a new file
+    :param input_path: path to netcdf containing the raw data
+    :param output_path: path to which the cleaned data should be written
+    :param config: config for the drought indicator
+    """
+
+    # rioxarray cannot open more than 3D datasets so use xarray
+    # the nc contains two bands, prob and C.Still not sure what C is
+    # used for but couldn't discover useful information in it and will
+    # give an error if trying to read both (cause C is also a variable
+    # in prob) the date format is formatted as months since 1960. In
+    # principle xarray can convert this type of data to datetime, but
+    # due to a wrong naming of the calendar variable it cannot do this
+    # automatically Thus first load with decode_times=False and then
+    # change the calendar variable and decode the months later
     iri_ds = xr.open_dataset(
-        IRI_filepath, decode_times=False, drop_variables="C"
+        input_path, decode_times=False, drop_variables="C"
```

Comment:
Given the experience with GloFAS data not opening properly with `xr.open_dataset()` maybe worth switching this to the same method that we've applied there? 

```
with xr.open_mfdataset(
                filepath_list,
                engine="cfgrib",
                backend_kwargs={
                    "indexpath": "",
                    "filter_by_keys": {"dataType": data_type},
                },
            ) as ds:
```

Reply:
Good point! Think in this case should be `xr.load_dataset()` (as it is `.nc` and not `.grib`) so changed it to that

Reply:
btw big disadvantage is of `load_dataset` that it can take a long time to load with larger datasets. With IRI it is still alright but e.g. with chirps it is not so nice. But dont know of a better solution :(